ext/pdo_dblib/dblib_stmt.c

```diff
@@ -203,8 +203,15 @@ static int pdo_dblib_stmt_describe(pdo_stmt_t *stmt, int colno TSRMLS_DC)
 	}
 	
 	struct pdo_column_data *col = &stmt->columns[colno];
-	
-	col->name = (char*)dbcolname(H->link, colno+1);
+
+	char *fname = (char*)dbcolname(H->link, colno+1);
+	char computed_buf[16];
+	if (fname && *fname) {
+		col->name = estrdup(fname);
+	} else {
+		snprintf(computed_buf,16,"computed%d", colno);
+		col->name = estrdup(computed_buf);
```

Comment:
You should ensure `col->name` is \0 terminated, strlen() is used afterwards. Or better, snprintf() returns the number of bytes written, that can also help to spare the strlen() call. Please consult man.

Thanks.